File: 6_kyu/PI_approximation.go
package kata

import (
	"math"
	"strconv"
)

func IterPi(epsilon float64) (int, string) {
	count := 1
	approx := float64(4)
	divisor := float64(-3)
	for math.Abs(math.Pi-approx) > epsilon {
		approx += 4. / divisor
		divisor = -(divisor + math.Copysign(2., divisor))
		count++
	}
	return count, strconv.FormatFloat(approx, 'f', 10, 64)
}

File: 6_kyu/PI_approximation.lua
local solution = {}

-- π approximation by Leibniz formula for π
-- Argument
--    delta to π to terminate the sum
-- Return
--    {number of iterations, approximation for π}
function solution.iterPi(epsilon)
  -- https://en.wikipedia.org/wiki/Leibniz_formula_for_%CF%80
  local approx = 0
  local i = 1
  local v = 4
  while math.abs(math.pi-approx) > epsilon do
    approx = approx + v/i
    i = i + 2
    v = -v
  end
  return {i//2, approx}
end
  
return solution
